packages/common/src/ContractInteractor.ts

```diff
@@ -1174,3 +1174,22 @@ export function getRawTxOptions (chainId: number, networkId: number, chain?: str
       }, 'london')
   }
 }
+
+export async function getChainId (provider: Web3ProviderBaseInterface): Promise<number> {
```

Comment:
Also, this global function has no business being declared in `ContractInteractor.ts`